app/scripts/services/fight.js

```diff
@@ -103,7 +42,7 @@ angular.module("rpgApp").service("FightEngine", ["CharServ", "AdversariesDB", "I
    *
    * @return {boolean} true if the player won the fight, false otherwise.
    */
-  function fightRound() {
+  function fightRound(opponents) {
```

Comment:
Because the function now has a list of opponents to take care of, and all opponents behave the same (they are `Fighter`s), you could iterate through that list and greatly reduce duplication.